[Bug] 无法在服务端运行

### OS Version - 操作系统

Windows

### Minecraft Version Details - Minecraft 版本信息

1.21 fabric-0.15.11

### Version Details - 版本信息

2.8.0+build.47

### Existing behavior - 现有行为

当在服务器安装当前适配1.21版本的mod时, 服务器启动失败. 使用代码库编译的最新版本也有此问题. 报错为
```
[14:47:38] [main/ERROR]: Uncaught exception in thread "main"
java.lang.RuntimeException: An exception occurred when launching the server!
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:71) ~[fabric-loader-0.15.11.jar:?]
        at net.fabricmc.installer.ServerLauncher.main(ServerLauncher.java:69) ~[fabric-server-mc.1.21-loader.0.15.11-launcher.1.0.1.jar:1.0.1]
Caused by: java.lang.RuntimeException: Failed to read accessWidener file from mod gca
        at net.fabricmc.loader.impl.FabricLoaderImpl.loadAccessWideners(FabricLoaderImpl.java:514) ~[fabric-loader-0.15.11.jar:?]
        at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:149) ~[fabric-loader-0.15.11.jar:?]
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68) ~[fabric-loader-0.15.11.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) ~[fabric-loader-0.15.11.jar:?]
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) ~[fabric-loader-0.15.11.jar:?]
        ... 1 more
Caused by: net.fabricmc.loader.impl.lib.accesswidener.AccessWidenerFormatException: Namespace (named) does not match current runtime namespace (intermediary)
        at net.fabricmc.loader.impl.lib.accesswidener.AccessWidenerReader.error(AccessWidenerReader.java:285) ~[fabric-loader-0.15.11.jar:?]
        at net.fabricmc.loader.impl.lib.accesswidener.AccessWidenerReader.read(AccessWidenerReader.java:84) ~[fabric-loader-0.15.11.jar:?]
        at net.fabricmc.loader.impl.FabricLoaderImpl.loadAccessWideners(FabricLoaderImpl.java:512) ~[fabric-loader-0.15.11.jar:?]
        at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:149) ~[fabric-loader-0.15.11.jar:?]
        at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68) ~[fabric-loader-0.15.11.jar:?]
        at net.fabricmc.loader.impl.launch.knot.KnotServer.main(KnotServer.java:23) ~[fabric-loader-0.15.11.jar:?]
        at net.fabricmc.loader.impl.launch.server.FabricServerLauncher.main(FabricServerLauncher.java:69) ~[fabric-loader-0.15.11.jar:?]
        ... 1 more
```

### Expected behavior - 预期行为

服务器应该正常启动

### To Reproduce - 复现问题

在服务端运行即可复现

### This issue is unique - 这个issue是唯一的

- [X] I have searched the issue tracker and did not find an issue describing my bug. - 我已经搜索问题追踪器并且没有发现相同的 issue。

--- a/src/main/java/dev/dubhe/gugle/carpet/mixin/EntityMixin.java
+++ b/src/main/java/dev/dubhe/gugle/carpet/mixin/EntityMixin.java
@@ -1,0 +1,25 @@
+package dev.dubhe.gugle.carpet.mixin;
+
+import dev.dubhe.gugle.carpet.GcaSetting;
+import net.minecraft.client.player.RemotePlayer;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.player.Player;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(Entity.class)
+public class EntityMixin {
+    @SuppressWarnings("ConstantValue")
+    @Inject(method = "interact", at = @At("HEAD"), cancellable = true)
+    private void interact(Player player, InteractionHand interactionHand, CallbackInfoReturnable<InteractionResult> cir) {
+        // 在客户端中，玩家可以与客户端的被交互玩家交互并返回PASS，这时交互玩家手上如果拿着可以使用的物品，则物品会被使用
+        // 所以如果判断被交互实体是客户端玩家，返回SUCCESS
+        if ((Object) this instanceof RemotePlayer && (GcaSetting.openFakePlayerInventory || GcaSetting.openFakePlayerEnderChest)) {
+            cir.setReturnValue(InteractionResult.SUCCESS);
+        }
+    }
+}

--- a/src/main/java/dev/dubhe/gugle/carpet/mixin/PlayerMixin.java
+++ b/src/main/java/dev/dubhe/gugle/carpet/mixin/PlayerMixin.java
@@ -9,7 +9,6 @@ import dev.dubhe.gugle.carpet.api.tools.text.ComponentTranslate;
 import dev.dubhe.gugle.carpet.tools.FakePlayerEnderChestContainer;
 import dev.dubhe.gugle.carpet.tools.FakePlayerInventoryContainer;
 import dev.dubhe.gugle.carpet.tools.FakePlayerInventoryMenu;
-import net.minecraft.client.player.RemotePlayer;
 import net.minecraft.world.InteractionHand;
 import net.minecraft.world.InteractionResult;
 import net.minecraft.world.SimpleMenuProvider;
@@ -44,10 +43,6 @@ abstract class PlayerMixin {
         if (entity instanceof EntityPlayerMPFake fakePlayer) {
             // 打开物品栏
             return this.openInventory(player, fakePlayer);
-        } else if (entity instanceof RemotePlayer) {
-            // 在客户端中，玩家可以与客户端的被交互玩家交互并返回PASS，这时交互玩家手上如果拿着可以使用的物品，则物品会被使用
-            // 所以如果判断被交互实体是客户端玩家，返回SUCCESS
-            return InteractionResult.SUCCESS;
         }
         return original.call(entity, player, hand);
     }

--- a/src/main/resources/gca.mixins.json
+++ b/src/main/resources/gca.mixins.json
@@ -23,6 +23,7 @@
   "server": [
   ],
   "client": [
+    "EntityMixin",
     "ServerPlayerMixin"
   ],
   "injectors": {

